builtin/providers/aws/resource_aws_waf_ipset.go

```diff
@@ -0,0 +1,175 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafIPSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafIPSetCreate,
+		Read:   resourceAwsWafIPSetRead,
+		Update: resourceAwsWafIPSetUpdate,
+		Delete: resourceAwsWafIPSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_set_descriptors": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafIPSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+
+	// ChangeToken
+	var ct *waf.GetChangeTokenInput
+	time.Sleep(10 * time.Second) // To Prevent WAFStaleDataException
```

Comment:
@stack72 if i don't wait for sometime i get this exception because of ChangeToken: res.ChangeToken i am not sure how to handle in a better way.